cloud_experts_tutorials/cloud-experts-getting-started/cloud-experts-getting-started-openshift-concepts.adoc

```diff
@@ -0,0 +1,88 @@
+
+. Start S2I a second time using the WAR file from the earlier step and the initial Tomcat image to create the runtime image.
+
+By placing build logic inside of images and combining the images into multiple steps, the runtime environment is close to the build environment without requiring the deployment of build tools to production.
+
+=== S2I benefits
+
+==== Reproducibility
+Allow build environments to be tightly versioned by encapsulating them within a container image and defining a simple interface of injected source code for callers. Reproducible builds are a key requirement for enabling security updates and continuous integration in containerized infrastructure, and builder images help ensure repeatability and the ability to swap run times.
+
+==== Flexibility
+Any existing build system that can run on Linux can run inside of a container, and each individual builder can also be part of a larger pipeline. The scripts that process the application source code can be injected into the builder image, allowing authors to adapt existing images to enable source handling.
+
+==== Speed
+Instead of building multiple layers in a single Dockerfile, S2I encourages authors to represent an application in a single image layer. This saves time during creation and deployment and allows for better control over the output of the final image.
+
+==== Security
+Dockerfiles are run without many of the normal operational controls of containers. They usually run as root and have access to the container network. S2I can control what permissions and privileges are available to the builder image since the build is launched in a single container. In concert with platforms like OpenShift, S2I allows administrators to control what privileges developers have at build time.
+
+== Routes
+An OpenShift Route exposes a service at a hostname so that external clients can reach it by name. When a Route object is created on OpenShift, it gets picked up by the built-in HAProxy load balancer to expose the requested service and make it externally available with the given configuration. Similar to the Kubernetes Ingress object, Red Hat created the concept of Route to fill a need and then contributed the design principles behind it to the community, which heavily influenced the Ingress design. A Route does have some additional features as can be seen in the following chart.
+
+image::ostoy-routes_ingress_comparison.png[Comparing routes and ingresses]
+
+[NOTE]
+====
+DNS resolution for a hostname is handled separately from routing. Your administrator might have configured a cloud domain that will always correctly resolve to the router or modify your unrelated hostname DNS records independently to resolve to the router.
+====
+
+An individual route can override some defaults by providing specific configurations in its annotations. See xref:../../networking/routes/route-configuration.adoc#nw-route-specific-annotations_route-configuration[route specific annotations] for more details.
```

Comment:
Nit on sentence form per [repo guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#lead-in-link-sentence-style-for-non-xref-situations):
```suggestion
An individual route can override some defaults by providing specific configurations in its annotations. For more information, see xref:../../networking/routes/route-configuration.adoc#nw-route-specific-annotations_route-configuration[Route-specific annotations].
```
or maybe better:
```suggestion
An individual route can override some defaults by providing specific configurations in its annotations.

[role="_additional-resources"]
.Additional resources
* xref:../../networking/routes/route-configuration.adoc#nw-route-specific-annotations_route-configuration[Route-specific annotations]
```